contracts/test/vanity_url.ts

```diff
@@ -0,0 +1,311 @@
+import { expect } from "chai";
+import { ethers, network } from "hardhat";
+import { BigNumber } from "ethers";
+import config from '../config'
+
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signers";
+
+import type { MockDC, VanityURL } from "../typechain-types"
+
+const maxWrapperExpiry = ethers.BigNumber.from(new Uint8Array(8).fill(255)).toString()
+const initConfiguration = {
+  wrapperExpiry: maxWrapperExpiry,
+  fuses: config.fuses,
+  registrarController: config.registrarController,
+  nameWrapper: config.nameWrapper,
+  baseRegistrar: config.registrar,
+  resolver: config.resolver,
+  reverseRecord: config.reverseRecord,
+  duration: config.duration
+}
+const dotName = 'test.1.country'
+const urlUpdatePrice = ethers.utils.parseEther("1");
+
+const increaseTime = async (sec: number): Promise<void> => {
+  await network.provider.send("evm_increaseTime", [sec]);
+  await network.provider.send("evm_mine");
+};
+
+const getTimestamp = async (): Promise<BigNumber> => {
+  const blockNumber = await ethers.provider.getBlockNumber();
+  const block = await ethers.provider.getBlock(blockNumber);
+  return BigNumber.from(block.timestamp);
+};
+
+describe('VanityURL', () => {
```

Comment:
I think we should snapshot and revert VM after each test, otherwise it will affect other tests using the same environment

Reply:
I deployed the contracts in beforeEach, so I think each test will not affect others.